File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/AiModel.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * A Data Lake AiModel details
* 
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=AiModel.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="modelType")

public final class AiModel extends Model {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
    @com.fasterxml.jackson.annotation.JsonProperty("id")
private String id;

public Builder id(String id) {
    this.id = id;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("modelName")
private String modelName;

public Builder modelName(String modelName) {
    this.modelName = modelName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("namespaceName")
private String namespaceName;

public Builder namespaceName(String namespaceName) {
    this.namespaceName = namespaceName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

public Builder description(String description) {
    this.description = description;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("createdBy")
private String createdBy;

public Builder createdBy(String createdBy) {
    this.createdBy = createdBy;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
private String updatedBy;

public Builder updatedBy(String updatedBy) {
    this.updatedBy = updatedBy;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
private LifecycleState lifecycleState;

public Builder lifecycleState(LifecycleState lifecycleState) {
    this.lifecycleState = lifecycleState;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleDetails")
private String lifecycleDetails;

public Builder lifecycleDetails(String lifecycleDetails) {
    this.lifecycleDetails = lifecycleDetails;
    return this;
}
            /**
     * Provides Ai Model's Capabilities.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("modelCapabilities")
private java.util.List<AiModelCapabilitiesEnum> modelCapabilities;

        /**
         * Provides Ai Model's Capabilities.
         * @param modelCapabilities the value to set
         * @return this builder
         **/
        

public Builder modelCapabilities(java.util.List<AiModelCapabilitiesEnum> modelCapabilities) {
    this.modelCapabilities = modelCapabilities;
    return this;
}
            /**
     * version that is available for that AI Model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("modelVersion")
private String modelVersion;

        /**
         * version that is available for that AI Model.
         * @param modelVersion the value to set
         * @return this builder
         **/
        

public Builder modelVersion(String modelVersion) {
    this.modelVersion = modelVersion;
    return this;
}
            /**
     * vendor name for that Model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("vendor")
private String vendor;

        /**
         * vendor name for that Model.
         * @param vendor the value to set
         * @return this builder
         **/
        

public Builder vendor(String vendor) {
    this.vendor = vendor;
    return this;
}
            /**
     * region source of that model
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("regionId")
private String regionId;

        /**
         * region source of that model
         * @param regionId the value to set
         * @return this builder
         **/
        

public Builder regionId(String regionId) {
    this.regionId = regionId;
    return this;
}


        public AiModel build() {
            AiModel model = new AiModel(this.id
                , this.modelName
                , this.displayName
                , this.namespaceName
                , this.description
                , this.timeCreated
                , this.timeUpdated
                , this.createdBy
                , this.updatedBy
                , this.lifecycleState
                , this.lifecycleDetails
                , this.modelCapabilities
                , this.modelVersion
                , this.vendor
                , this.regionId);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(AiModel model) {
                this.id(model.getId());
    this.modelName(model.getModelName());
    this.displayName(model.getDisplayName());
    this.namespaceName(model.getNamespaceName());
    this.description(model.getDescription());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.createdBy(model.getCreatedBy());
    this.updatedBy(model.getUpdatedBy());
    this.lifecycleState(model.getLifecycleState());
    this.lifecycleDetails(model.getLifecycleDetails());
    this.modelCapabilities(model.getModelCapabilities());
    this.modelVersion(model.getModelVersion());
    this.vendor(model.getVendor());
    this.regionId(model.getRegionId());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public AiModel(String id, String modelName, String displayName, String namespaceName, String description, java.util.Date timeCreated, java.util.Date timeUpdated, String createdBy, String updatedBy, LifecycleState lifecycleState, String lifecycleDetails, java.util.List<AiModelCapabilitiesEnum> modelCapabilities, String modelVersion, String vendor, String regionId) {
    super(id, modelName, displayName, namespaceName, description, timeCreated, timeUpdated, createdBy, updatedBy, lifecycleState, lifecycleDetails);
        this.modelCapabilities = modelCapabilities;
        this.modelVersion = modelVersion;
        this.vendor = vendor;
        this.regionId = regionId;
    }


        /**
     * Provides Ai Model's Capabilities.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("modelCapabilities")
    private final java.util.List<AiModelCapabilitiesEnum> modelCapabilities;

        /**
     * Provides Ai Model's Capabilities.
     * @return the value
     **/
    
    public java.util.List<AiModelCapabilitiesEnum> getModelCapabilities() {
        return modelCapabilities;
    }


        /**
     * version that is available for that AI Model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("modelVersion")
    private final String modelVersion;

        /**
     * version that is available for that AI Model.
     * @return the value
     **/
    
    public String getModelVersion() {
        return modelVersion;
    }


        /**
     * vendor name for that Model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("vendor")
    private final String vendor;

        /**
     * vendor name for that Model.
     * @return the value
     **/
    
    public String getVendor() {
        return vendor;
    }


        /**
     * region source of that model
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("regionId")
    private final String regionId;

        /**
     * region source of that model
     * @return the value
     **/
    
    public String getRegionId() {
        return regionId;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("AiModel(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", modelCapabilities=").append(String.valueOf(this.modelCapabilities));
        sb.append(", modelVersion=").append(String.valueOf(this.modelVersion));
        sb.append(", vendor=").append(String.valueOf(this.vendor));
        sb.append(", regionId=").append(String.valueOf(this.regionId));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AiModel)) {
            return false;
        }

        AiModel other = (AiModel) o;
        return java.util.Objects.equals(this.modelCapabilities, other.modelCapabilities) &&
            java.util.Objects.equals(this.modelVersion, other.modelVersion) &&
            java.util.Objects.equals(this.vendor, other.vendor) &&
            java.util.Objects.equals(this.regionId, other.regionId) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.modelCapabilities == null ? 43 : this.modelCapabilities.hashCode());
        result = (result * PRIME) + (this.modelVersion == null ? 43 : this.modelVersion.hashCode());
        result = (result * PRIME) + (this.vendor == null ? 43 : this.vendor.hashCode());
        result = (result * PRIME) + (this.regionId == null ? 43 : this.regionId.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/ai_model.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.


from .model import Model
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class AiModel(Model):
    """
    A Data Lake AiModel details
    """

    def __init__(self, **kwargs):
        """
        Initializes a new AiModel object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.AiModel.model_type` attribute
        of this class is ``GEN_AI`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param model_type:
            The value to assign to the model_type property of this AiModel.
            Allowed values for this property are: "GEN_AI"
        :type model_type: str

        :param id:
            The value to assign to the id property of this AiModel.
        :type id: str

        :param model_name:
            The value to assign to the model_name property of this AiModel.
        :type model_name: str

        :param display_name:
            The value to assign to the display_name property of this AiModel.
        :type display_name: str

        :param namespace_name:
            The value to assign to the namespace_name property of this AiModel.
        :type namespace_name: str

        :param description:
            The value to assign to the description property of this AiModel.
        :type description: str

        :param time_created:
            The value to assign to the time_created property of this AiModel.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this AiModel.
        :type time_updated: datetime

        :param created_by:
            The value to assign to the created_by property of this AiModel.
        :type created_by: str

        :param updated_by:
            The value to assign to the updated_by property of this AiModel.
        :type updated_by: str

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this AiModel.
            Allowed values for this property are: "ACTIVE", "CREATING", "DELETING", "DELETED", "FAILED"
        :type lifecycle_state: str

        :param lifecycle_details:
            The value to assign to the lifecycle_details property of this AiModel.
        :type lifecycle_details: str

        :param model_capabilities:
            The value to assign to the model_capabilities property of this AiModel.
        :type model_capabilities: list[oci.aidataplatform_dp.models.AiModelCapabilitiesEnum]

        :param model_version:
            The value to assign to the model_version property of this AiModel.
        :type model_version: str

        :param vendor:
            The value to assign to the vendor property of this AiModel.
        :type vendor: str

        :param region_id:
            The value to assign to the region_id property of this AiModel.
        :type region_id: str

        """
        self.swagger_types = {
            'model_type': 'str',
            'id': 'str',
            'model_name': 'str',
            'display_name': 'str',
            'namespace_name': 'str',
            'description': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'created_by': 'str',
            'updated_by': 'str',
            'lifecycle_state': 'str',
            'lifecycle_details': 'str',
            'model_capabilities': 'list[AiModelCapabilitiesEnum]',
            'model_version': 'str',
            'vendor': 'str',
            'region_id': 'str'
        }

        self.attribute_map = {
            'model_type': 'modelType',
            'id': 'id',
            'model_name': 'modelName',
            'display_name': 'displayName',
            'namespace_name': 'namespaceName',
            'description': 'description',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'created_by': 'createdBy',
            'updated_by': 'updatedBy',
            'lifecycle_state': 'lifecycleState',
            'lifecycle_details': 'lifecycleDetails',
            'model_capabilities': 'modelCapabilities',
            'model_version': 'modelVersion',
            'vendor': 'vendor',
            'region_id': 'regionId'
        }

        self._model_type = None
        self._id = None
        self._model_name = None
        self._display_name = None
        self._namespace_name = None
        self._description = None
        self._time_created = None
        self._time_updated = None
        self._created_by = None
        self._updated_by = None
        self._lifecycle_state = None
        self._lifecycle_details = None
        self._model_capabilities = None
        self._model_version = None
        self._vendor = None
        self._region_id = None
        self._model_type = 'GEN_AI'

    @property
    def model_capabilities(self):
        """
        Gets the model_capabilities of this AiModel.
        Provides Ai Model's Capabilities.


        :return: The model_capabilities of this AiModel.
        :rtype: list[oci.aidataplatform_dp.models.AiModelCapabilitiesEnum]
        """
        return self._model_capabilities

    @model_capabilities.setter
    def model_capabilities(self, model_capabilities):
        """
        Sets the model_capabilities of this AiModel.
        Provides Ai Model's Capabilities.


        :param model_capabilities: The model_capabilities of this AiModel.
        :type: list[oci.aidataplatform_dp.models.AiModelCapabilitiesEnum]
        """
        self._model_capabilities = model_capabilities

    @property
    def model_version(self):
        """
        Gets the model_version of this AiModel.
        version that is available for that AI Model.


        :return: The model_version of this AiModel.
        :rtype: str
        """
        return self._model_version

    @model_version.setter
    def model_version(self, model_version):
        """
        Sets the model_version of this AiModel.
        version that is available for that AI Model.


        :param model_version: The model_version of this AiModel.
        :type: str
        """
        self._model_version = model_version

    @property
    def vendor(self):
        """
        Gets the vendor of this AiModel.
        vendor name for that Model.


        :return: The vendor of this AiModel.
        :rtype: str
        """
        return self._vendor

    @vendor.setter
    def vendor(self, vendor):
        """
        Sets the vendor of this AiModel.
        vendor name for that Model.


        :param vendor: The vendor of this AiModel.
        :type: str
        """
        self._vendor = vendor

    @property
    def region_id(self):
        """
        Gets the region_id of this AiModel.
        region source of that model


        :return: The region_id of this AiModel.
        :rtype: str
        """
        return self._region_id

    @region_id.setter
    def region_id(self, region_id):
        """
        Sets the region_id of this AiModel.
        region source of that model


        :param region_id: The region_id of this AiModel.
        :type: str
        """
        self._region_id = region_id

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
